frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.unit.spec.tsx

```diff
@@ -35,19 +43,30 @@ function getTableObject(table: Table) {
 
 type SetupOpts = {
   action?: WritebackQueryAction;
+  storeInitialState?: Partial<State>;
```

Comment:
Think the preferred way to setup state is to assemble its final shape inside `setup` and expose arguments to tweak important stuff, e.g.:

```jsx
function setup({ isAdmin, isSharingEnabled }) {
  const currentUser = createMockUser({ is_superuser: isAdmin })
  const settings = createMockSettingsState({ "enable-public-sharing": isSharingEnabled  })
  renderWithProviders(<JSX />, {
    storeInitialState: createMockState({ currentUser, settings })
  });
}
```

Reply:
I could do that. It just feels like if we keep cramping more and more flags to the `setup` it's gonna be unbearable to read. That's why I chose to do this since I don't want to share much logic with other tests.

I'll try to put this logic inside `setup` and see how it becomes.

Reply:
I think if it comes down to it, than it's a potential sign that we need to decompose the component

Reply:
Doing this isn't too bad tbh. I've followed your suggestion.